float64 pos1
float64 pos2
float64 pos3
float64 orient1
float64 orient2
float64 orient3
float64 orient4
